File: sorting/src/main/java/com/sorting/insertionsort/InsertionSort.py
def issortingrequired(elementfromlowerindex,elementfromheigherindex):
    return elementfromlowerindex > elementfromheigherindex;


def swappingtheelements(elements,fromindex,toindex):
    temp = elements[fromindex]
    elements[fromindex] = elements[toindex]
    elements[toindex] = temp


def printingtheelements(elements):
    for element in range(0,len(elements)):
        print(elements[element])


def insertionsorting(elements):
    position = 0
    for i in range(0,len(elements)-1):
       position=i
       for position in range(position,-1,-1): 
        if(issortingrequired(elements[position],elements[position+1])):
            swappingtheelements(elements,position,position+1)
    return elements

if __name__ == "__main__":
    elements = [ 5, 1, 4, 8, 2 ]
    listofelements = insertionsorting(elements) 
    printingtheelements(listofelements)                

File: sorting/src/main/java/com/sorting/insertionsort/InsertionSort.js
function isSortingRequired(elementfromLowerIndex, elementFromHeigherIndex) {
    return elementfromLowerIndex > elementFromHeigherIndex
}

function swappingtheelements(elements, fromIndex, toIndex) {
    var temp = elements[fromIndex]
    elements[fromIndex] = elements[toIndex]
    elements[toIndex] = temp
}

function insertionSort(elements) {
    for (let i = 0; i < elements.length; i++) {
        var position = i
        for (let j = position; (j < (elements.length - 1) && j >= 0); j--) {
            if (isSortingRequired(elements[j], elements[j + 1])) {
                swappingtheelements(elements, j, j + 1)
            }
        }
    }
    return elements
}

function printOutElements(elements) {
    elements.forEach(element => {
        console.log(element)
    });
}

var elementsNeedToSort = [5, 1, 4, 8, 2]
var elementsSort = insertionSort(elementsNeedToSort)
printOutElements(elementsSort)
